File: resource-registry/src/test/java/nl/haploid/octowight/registry/repository/ResourceModelDmoRepositoryIT.java
package nl.haploid.octowight.registry.repository;

import nl.haploid.octowight.registry.AbstractIT;
import nl.haploid.octowight.registry.TestData;
import org.junit.Test;
import org.springframework.beans.factory.annotation.Autowired;

import static org.junit.Assert.assertEquals;

public class ResourceModelDmoRepositoryIT extends AbstractIT {

	@Autowired
	private ResourceModelDmoRepository resourceModelDmoRepository;

	@Test
	public void testFindOne() {
		// TODO: move to TestData
		final ResourceModelDmo expectedResourceModelDmo = new ResourceModelDmo();
		final ResourceModelId resourceModelId = new ResourceModelId();
		resourceModelId.setResourceId(TestData.nextLong());
		resourceModelId.setResourceType(TestData.nextString());
		expectedResourceModelDmo.setId(resourceModelId);
		expectedResourceModelDmo.setBody(TestData.nextString());
		resourceModelDmoRepository.save(expectedResourceModelDmo);
		final ResourceModelDmo actualResourceModelDmo = resourceModelDmoRepository.findOne(resourceModelId);
		assertEquals(expectedResourceModelDmo.getBody(), actualResourceModelDmo.getBody());
	}
}

File: resource-registry/src/test/scala/nl/haploid/octowight/registry/repository/ResourceModelDmoRepositoryIT.scala
package nl.haploid.octowight.registry.repository

import nl.haploid.octowight.registry.{AbstractIT, TestData}
import org.springframework.beans.factory.annotation.Autowired

class ResourceModelDmoRepositoryIT extends AbstractIT {
  @Autowired private[this] val resourceModelDmoRepository: ResourceModelDmoRepository = null

  behavior of "Resource model repository"

  it should "find one" in {
    val expectedResourceModelDmo = TestData.resourceModelDmo
    resourceModelDmoRepository.save(expectedResourceModelDmo)
    val actualResourceModelDmo = resourceModelDmoRepository.findOne(expectedResourceModelDmo.id)
    actualResourceModelDmo.body should be(expectedResourceModelDmo.body)
  }
}
